clusters/resource_cluster.go

```diff
@@ -57,71 +58,119 @@ func ZoneDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
-			func(ss map[string]*schema.Schema) map[string]*schema.Schema {
-				ss["library"].Set = func(i any) int {
-					lib := libraries.NewLibraryFromInstanceState(i)
-					return schema.HashString(lib.String())
-				}
-				return ss
-			})["library"]
+func (ClusterResourceProvider) Aliases() map[string]string {
+	return map[string]string{"cluster_mount_infos": "cluster_mount_info"}
+}
 
-		s["autotermination_minutes"].Default = 60
-		s["cluster_id"] = &schema.Schema{
+func (ClusterResourceProvider) CustomizeSchema(s map[string]*schema.Schema) map[string]*schema.Schema {
+	common.CustomizeSchemaPath(s, "enable_elastic_disk").SetComputed()
+	common.CustomizeSchemaPath(s, "enable_local_disk_encryption").SetComputed()
+	common.CustomizeSchemaPath(s, "node_type_id").SetComputed().SetConflictsWith([]string{"driver_instance_pool_id", "instance_pool_id"})
+	common.CustomizeSchemaPath(s, "driver_node_type_id").SetComputed().SetConflictsWith([]string{"driver_instance_pool_id", "instance_pool_id"})
+	common.CustomizeSchemaPath(s, "driver_instance_pool_id").SetComputed().SetConflictsWith([]string{"driver_node_type_id", "node_type_id"})
+	common.CustomizeSchemaPath(s, "ssh_public_keys").SetMaxItems(10)
+	common.CustomizeSchemaPath(s, "init_scripts").SetMaxItems(10).AddNewField("abfss", common.StructToSchema(InitScriptStorageInfo{}, nil)["abfss"]).AddNewField("gcs", common.StructToSchema(InitScriptStorageInfo{}, nil)["gcs"])
+	common.CustomizeSchemaPath(s, "init_scripts", "dbfs").SetDeprecated(DbfsDeprecationWarning)
+	common.CustomizeSchemaPath(s, "init_scripts", "dbfs", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "init_scripts", "s3", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "init_scripts", "volumes", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "init_scripts", "workspace", "destination").SetRequired()
+	common.CustomizeSchemaPath(s, "workload_type", "clients").SetRequired()
+	common.CustomizeSchemaPath(s, "workload_type", "clients", "notebooks").SetDefault(true)
+	common.CustomizeSchemaPath(s, "workload_type", "clients", "jobs").SetDefault(true)
+	common.CustomizeSchemaPath(s).AddNewField("idempotency_token", &schema.Schema{
+		Type:     schema.TypeString,
+		Optional: true,
+		ForceNew: true,
+	})
+	common.CustomizeSchemaPath(s, "data_security_mode").SetSuppressDiff()
+	common.CustomizeSchemaPath(s, "docker_image", "url").SetRequired()
+	common.CustomizeSchemaPath(s, "docker_image", "basic_auth", "password").SetRequired().SetSensitive()
+	common.CustomizeSchemaPath(s, "docker_image", "basic_auth", "username").SetRequired()
+	common.CustomizeSchemaPath(s, "spark_conf").SetCustomSuppressDiff(SparkConfDiffSuppressFunc)
+	common.CustomizeSchemaPath(s, "aws_attributes").SetSuppressDiff().SetConflictsWith([]string{"azure_attributes", "gcp_attributes"})
+	common.CustomizeSchemaPath(s, "aws_attributes", "zone_id").SetCustomSuppressDiff(ZoneDiffSuppress)
```

Comment:
We have few more attributes in the new schema - we need to at least document them:

```
@ ["provider_schemas","registry.terraform.io/databricks/databricks","resource_schemas","databricks_cluster","block","attributes","cluster_source"]
+ {"description_kind":"plain","optional":true,"type":"string"}
@ ["provider_schemas","registry.terraform.io/databricks/databricks","resource_schemas","databricks_cluster","block","block_types","aws_attributes","block","attributes","ebs_volume_iops"]
+ {"description_kind":"plain","optional":true,"type":"number"}
@ ["provider_schemas","registry.terraform.io/databricks/databricks","resource_schemas","databricks_cluster","block","block_types","aws_attributes","block","attributes","ebs_volume_throughput"]
+ {"description_kind":"plain","optional":true,"type":"number"}
@ ["provider_schemas","registry.terraform.io/databricks/databricks","resource_schemas","databricks_cluster","block","block_types","azure_attributes","block","block_types"]
+ {"log_analytics_info":{"block":{"attributes":{"log_analytics_primary_key":{"description_kind":"plain","optional":true,"type":"string"},"log_analytics_workspace_id":{"description_kind":"plain","optional":true,"type":"string"}},"description_kind":"plain"},"max_items":1,"nesting_mode":"list"}}
```

Reply:
This is interesting.

- `cluster_source` Feels like this shouldn't be specified as an input argument, I asked in the ticket I filed for the cluster team
- `ebs_volume_iops`, `ebs_volume_throughput` and `log_analytics_info` are not documented on the REST API documentation as well. I also asked in that ticket

Reply:
We'll mark cluster_source as read-only/computed and make a follow-up to document the rest.